### PR TITLE
Shadow pipelines

### LINKED ISSUE #1889: Add Parallel Traffic Splitting / Shadow Pipelines / Mirroring
### Description

Consider adding parallel traffic routing in order to create shadow pipelines. A shadow pipeline consists of having 2 models running in parallel, one serving the requests while the other does some other task, while not serving the request (e.g. model evaluation to keep track of the performance)

### Motivation

Can be useful for complex MLOps pipelines to keep track of model performance.

### Additional context

* [Istio mirroring docs](https://istio.io/latest/docs/tasks/traffic-management/mirroring/)
* [ingress-nginx mirroring docs](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#mirror)
* Seldon supports this feature, through ingress routing (see [here](https://docs.seldon.io/projects/seldon-core/en/latest/examples/ambassador_shadow.html)).

### Checklist

- [ ]  Can you have multiple primaries and a shadow pipeline? What are the possible combinations of multiple primaries and shadow pipelines? Should we support all combinations or only a few to maintain a nice user experience?
- [ ]  Add a field to indicate which branch will be the primary and which branch(es?) will be mirror
- [ ]  Add validations for the designed UI
- [ ]  Update telemetry and UserStr functions accordingly
- [ ]  Update docs

### PR DESCRIPTION
**Pull request by vishalbollu**
closes #1889 

---

checklist:

- [ ] run `make test` and `make lint`
- [ ] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

**Review by RobertLucian — APPROVED** [review]
LGTM!

**Review by miguelvr — COMMENTED** [review]
could you add a test API in test/apis ?

File: pkg/types/spec/validations.go [hunk 1]
```diff
@@ -846,6 +850,16 @@ func ValidateTrafficSplitter(api *userconfig.API) error {
 		return err
 	}
 
+	isShadow := false
```
Comment (deliahu) [inline pkg/types/spec/validations.go:853]: nit: `isShadow` -> `hasShadow`

File: test/apis/traffic-splitter/request_recorder.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class PythonPredictor:
+    def __init__(self, config):
+        pass
+
+    def predict(self, payload):
+        print(payload)
```
Comment (deliahu) [inline test/apis/traffic-splitter/request_recorder.py:6]: nit: consider using our logger

**Review by deliahu — APPROVED** [review]

**Pull request merged**